### PR TITLE
Avro format improvements

### PR DESCRIPTION
**Pull request by oandrew**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Detailed description / Documentation draft:

Continuation of #8717 
- References to previously defined names
- Allow AVRO_FIXED -> CH String
- Add Avro documentation

### CODE REVIEW

File: dbms/src/Processors/Formats/Impl/AvroRowInputFormat.h [hunk 1]
```diff
@@ -27,8 +28,9 @@ class AvroDeserializer
 private:
     using DeserializeFn = std::function<void(IColumn & column, avro::Decoder & decoder)>;
     using SkipFn = std::function<void(avro::Decoder & decoder)>;
+    struct SkipFnHolder { SkipFn skip_fn; };
```
Comment (alexey-milovidov) [inline dbms/src/Processors/Formats/Impl/AvroRowInputFormat.h]: Don't understand why we need this.
Everything else is Ok.

Comment (oandrew) [reply dbms/src/Processors/Formats/Impl/AvroRowInputFormat.h]: You are right. Got rid of it

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Ok.

PS. No idea why performance test failed badly. Probably some server from our CI fleet was overheated :) Will look at the results on next commits.

**Comment by tavplubix** [comment]
Backport to 20.1 not required

**Pull request merged**